### ISSUE TITLE
Avoid duplicated generated code when splitting dataset

### ISSUE THREAD

**Issue by felipealencar** [kate-holdener/ml_code_generator#50]
When a user executes the **Split Data** feature more than once, the code generation repeats the `X.sample` and `X.drop` lines. This behavior might not be desired in the final Python code.

Possible solutions:
1. Implement an application state to check if code was generated with the split method already.
2. Implement a _find and update_ method at the `CodeBlock` class.
3. Simply clear the `CodeBlock` state after each code generation.

Example:
![Captura de Tela 2022-10-09 às 01 17 02](https://user-images.githubusercontent.com/3163733/194737769-ac5cc6ac-40aa-40a0-9875-fa84239b6d18.png)

**Comment by kate-holdener** [comment]
Indeed! In fact, for any duplicate "next action", the user chooses, duplicated code will be generated, so I think this issue is more general than the **Split Data** feature. 

I think we either need to filter generated code, to remove duplication OR disallow the user to do duplicated actions (somewhat related to issue #3). If we go with disallowing the user to do duplicated actions, we'd need to add an "Undo" option, to allow the user to go back to previous steps.

**Comment by kate-holdener** [comment]
Another possible solution: update CodeBlock (in model/code_blocks.py) to have an "id" associated with it. When new block is added by calling add_next_block in AllBlocks, if a block with the same id already exists in AllBlocks, the old version of the block should be removed and the new one added to the end. AllBlocks::add_next_block is called from application/code_generator.py _create_new_block function, which gets called from _parse_and_execute. We can use the 'template' name passed to _parse_and_execute to generate the id: keep a dictionary that maps a template name to a unique id. If the template name is not in the dictionary, add it with a new unique id.
